node/modules/lp2p.go

```diff
@@ -2,15 +2,21 @@ package modules
 
 import (
 	"github.com/filecoin-project/boost/build"
+	"github.com/filecoin-project/boost/node/config"
 	"github.com/filecoin-project/lotus/node/modules/lp2p"
 	"github.com/libp2p/go-libp2p"
 )
 
-func simpleOpt(opt libp2p.Option) func() (opts lp2p.Libp2pOpts, err error) {
+func UserAgent(cfg *config.Boost) func() (opts lp2p.Libp2pOpts, err error) {
```

Comment:
We put the fullnode api into DI:

https://github.com/filecoin-project/boost/blob/543f878f0c85bc1d55048a2d54f6be225dcd0370/cmd/boostd/run.go#L130

So we can just use it here like so:
```suggestion
func UserAgent func(api v1api.FullNode) (opts lp2p.Libp2pOpts, err error) {
    v, err := api.Version(context.TODO())
```